cstr/src/lib.rs

```diff
@@ -180,6 +181,77 @@ pub unsafe fn write_to_buffer(message: &str, len: usize, output_ptr: *mut WideCh
     v.len() as u16
 }
 
+///
+/// write_wstring_slice_to_buffer writes the input WideChar slice (wstring) to the output buffer, and returns the number of bytes written
+///
+/// # Safety
+/// This writes to a raw c-pointer, which requires unsafe operations
+pub unsafe fn write_wstring_slice_to_buffer(
+    message: &[WideChar],
+    len: usize,
+    output_ptr: *mut WideChar,
+) -> u16 {
```

Comment:
We should preallocate this Vec so that the push of 0 doesn't cause a realloc

You can do this using Vec::with_capacity:
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity

and extend_from_slice: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice

Reply:
Honestly though, a better option would be to not copy to the `Vec` at all, just write the slice to the pointer, and then put a 0 in `offeset(len_of_messsage)`.